### ISSUE TITLE
Using Swift Package Manager instead of CocoaPods results in failed build

### ISSUE THREAD

**Issue by ryansommers** [rnmapbox/maps#2943]
## Environment
macOS 13.4.1
Xcode 14.3.1
React Native 0.67.4

## Steps to reproduce
1. Open rnmapbox example project

2. Add 
```
$RNMapboxMapsSwiftPackageManager = {
  url: "https://github.com/mapbox/mapbox-maps-ios.git",
  requirement: {
    kind: "exactVersion",
    version: "10.13.1"
  },
  product_name: "MapboxMaps"
}
```
to the Podfile, or 
```
$RNMapboxMapsSwiftPackageManager = 'manual'
```
and add the package manually to the project in Xcode.

3. Build in Xcode, fails with `No such module 'MapboxMaps'`

```sh   
cd example/ios
pod install
cd ../
react-native run-ios
```

Am I missing a step somewhere to properly use Mapbox through SPM instead of Pods?

-----

Adding a component to try and get around the bot auto closing, though it's not part of the "Setup/installation error" bug template:

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
} from '@rnmapbox/maps';

const aLine = {
  type: 'LineString',
  coordinates: [
    [-74.00597, 40.71427],
    [-74.00697, 40.71527],
  ],
};

class BugReportExample extends React.Component {
  render() {
    return (
      <MapView style={{flex: 1}}>
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id="idStreetLayer" shape={aLine}>
          <LineLayer id="idStreetLayer" />
        </ShapeSource>
      </MapView>
    );
  }
}
```

**Comment by ryansommers** [comment]
Adding code example as required:
```sh
cd example/ios
pod install
cd ../
react-native run-ios
```

**Comment by mfazekas** [comment]
@ryansommers What is the use case for SPM? 

I think it's because of an Xcode issue if an SPM is added to non main project.  You'll need to modify `Swift Import Path` and add `${SYMROOT}/${CONFIGURATION}${EFFECTIVE_PLATFORM_NAME}/`.
See step 3 at https://github.com/mfazekas/pods-spm-dep-poc/#how-it-works

**Comment by ryansommers** [comment]
@mfazekas I have Mapbox Navigation in SPM, and when I build with that and Mapbox Maps as a Pod (v10 only, v8 was working fine), I get “XXX-XXX.bcsymbolmap” couldn’t be copied to “BCSymbolMaps” because an item with the same name already exists."

I haven't been able to find a workaround for that, so was trying to use Maps in SPM as a workaround. 

And I have Mapbox Navigation in SPM because if it's installed as a Pod, there's incorrect languages that show up in various areas, unless `use_frameworks!` is added, which I can't do at the moment.

I'll give your suggestion a try shortly, thanks!

**Comment by ryansommers** [comment]
@mfazekas Just want to make sure I did this right:

1. Clicked Pods in the Project navigator
2. Clicked Pods under Project
3. Added MapboxMaps as a SPM dependency, and chose "rnmapbox-maps" under "Add to Target" 
4. Selected "rnmapbox-maps" under Targets and added `${SYMROOT}/${CONFIGURATION}${EFFECTIVE_PLATFORM_NAME}/` to Swift Import Paths
 
The project built and the "No such module" went away, but now I'm getting the error `21940 duplicate symbols for architecture x86_64`. I'm just guessing, but that's probably due to MapboxMaps being a dependency of MapboxNavigation, so now there's duplicates?
